store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java

```diff
@@ -176,17 +151,44 @@ public Boolean call(Key barCode) {
      */
     @Override
     public Observable<Parsed> disk(@Nonnull final Key barCode) {
+        if (shouldReturnNetworkBeforeStale(persister, stalePolicy, barCode)) {
+            return Observable.empty();
+        }
+
+        return readDisk(barCode);
+    }
+
+    private Observable<Parsed> readDisk(@Nonnull final Key barCode) {
         return persister().read(barCode)
                 .onErrorResumeNext(new OnErrorResumeWithEmpty<Raw>())
                 .map(parser)
                 .doOnNext(new Action1<Parsed>() {
                     @Override
                     public void call(Parsed parsed) {
                         updateMemory(barCode, parsed);
+                        if (stalePolicy == StalePolicy.REFRESH_ON_STALE
+                                && persisterIsStale(barCode, persister)) {
+                            backfillCache(barCode);
+                        }
                     }
                 }).cache();
     }
 
+    private void backfillCache(@Nonnull Key barCode) {
+        fetch(barCode).subscribe(new Action1<Parsed>() {
+            @Override
+            public void call(Parsed parsed) {
+                //do Nothing we are just backfilling cache
+            }
+        }, new Action1<Throwable>() {
+            @Override
+            public void call(Throwable throwable) {
+                //do nothing as we are just backfilling cache
+            }
+        });
```

Comment:
Maybe create a "noop" `Action1` to use here for readability instead of the empty / comment implementations. Could also be reused if needed elsewhere

Ex.
```
        Action1 noop = new Action1(){
            @Override
            public void call(Object o) {
            // do nothing
            }
        };
...
        fetch(barCode).subscribe(noop, noop);
```